agent/recorder.go

```diff
@@ -64,6 +64,10 @@ func NewSpanRecorder(agent *Agent) *SpanRecorder {
 func (r *SpanRecorder) RecordSpan(span tracer.RawSpan) {
 	r.Lock()
 	defer r.Unlock()
+	if !r.t.Alive() {
+		r.logger.Printf("an span is received but the recorder is already disposed.\n")
+		return
+	}
```

Comment:
We can remove this IMHO

Reply:
With this, we detect there is no loop function running so we don't append to the slice of spans. Maybe we can remove the logger line, but the if and return helps in allocations.